### PR TITLE
Move workers/datasets_based to services/worker

### PR DESCRIPTION
**Pull request by severo**
Based on #792.

### CODE REVIEW

File: services/worker/pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [tool.poetry]
 authors = ["Sylvain Lesage <sylvain.lesage@huggingface.co>"]
-description = "Worker for processing steps that need the datasets library"
-name = "datasets_based"
+description = "Worker that processes jobs and store the responses in the cache"
```
Comment (AndreaFrancis) [inline services/worker/pyproject.toml:3]:
```suggestion
description = "Worker that processes jobs and stores the responses in the cache"
```

File: services/worker/src/worker/config.py [hunk 2]
```diff
@@ -24,15 +24,15 @@ def get_empty_str_list() -> List[str]:
...
     max_load_pct: int = WORKER_LOOP_MAX_LOAD_PCT
     max_memory_pct: int = WORKER_LOOP_MAX_MEMORY_PCT
     sleep_seconds: int = WORKER_LOOP_SLEEP_SECONDS
     storage_paths: List[str] = field(default_factory=get_empty_str_list)
 
     @classmethod
-    def from_env(cls) -> "WorkerLoopConfig":
+    def from_env(cls) -> "LoopConfig":
         env = Env(expand_vars=True)
         with env.prefixed("WORKER_LOOP_"):
```
Comment (AndreaFrancis) [inline services/worker/src/worker/config.py:37]:
```suggestion
        with env.prefixed("LOOP_"):
```

Comment (AndreaFrancis) [reply services/worker/src/worker/config.py:37]: But I am not sure, we might need to change all other WORKER_ configs as well :/. Maybe keep it as it is

**Review by AndreaFrancis — COMMENTED** [review]
I put comments for minor changes. Just one question, how do we use that new vendor trec-car?

**Comment by severo** [comment]
> how do we use that new vendor trec-car?

it's not new, it's just moved with the rest of the code. It's a dependency required by some dataset scripts on the Hub: 
https://github.com/huggingface/datasets-server/blob/5267f427dfa5bae2e4b4fba0b8042d4783fc3a39/services/worker/pyproject.toml#L39
I had to put it inside the code because it's not available in pypi or github or anything (as far as I could find).

File: services/worker/src/worker/config.py (hunk 2 above)
Comment (severo) [reply services/worker/src/worker/config.py]: You're right, I have to change in the rest of the code

Comment (severo) [reply services/worker/src/worker/config.py:37]: well, I will let as WORKER_LOOP better in the code as it's more descriptive. It's not coherent, maybe we should let worker_loop.py in the code...

**Review by AndreaFrancis — APPROVED** [review]

**Pull request merged**